src/sensor/hexvalidator.cpp

```diff
@@ -31,7 +31,7 @@ bool HexValidator::check(const QByteArray& bytes)
     }
 
     // Check if it has an odd number of elements
-    if(bytes.size()%2 == 0) {
+    if(bytes.size() & 0x1) {
```

Comment:
Don't try to be smarter than the compiler ;)
![image](https://user-images.githubusercontent.com/1215497/61805977-3c3dc600-ae0d-11e9-94af-000285e65cdd.png)